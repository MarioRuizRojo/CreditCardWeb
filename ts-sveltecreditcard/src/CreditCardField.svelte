<script lang="ts">
  /**
   * name of the field of the form
   */
  export let fieldName: string;
  /**
   * css class of the icon to show in this field
   */
  export let filedIconClass: string;
  /**
   * text to show in the input field when empty
   */
  export let fieldPlaceHolder: string;
  /**
   * text value of the field
   */
  export let fieldValue: string;
  /**
   * text to show when validation fails
   */
  export let fieldError: string;
  /**
   * function to execute when user blur this input
   */
  export let fieldBlur: () => void;
</script>

<div
  data-testid={"maindiv" + fieldName}
  class="input-group input-group-lg mb-3"
>
  <span class="input-group-text">
    <i data-testid={"icon" + fieldName} class={filedIconClass} />
  </span>
  <input
    data-testid={"input" + fieldName}
    type="text"
    name={fieldName}
    class="form-control"
    placeholder={fieldPlaceHolder}
    bind:value={fieldValue}
    on:blur={fieldBlur}
  />
  {#if fieldError != ""}
    <div data-testid={"errorDiv" + fieldName} style="color:red">
      {fieldError}
    </div>
  {/if}
</div>
