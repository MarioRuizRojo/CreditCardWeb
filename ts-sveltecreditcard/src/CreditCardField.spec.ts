import CreditCardField from './CreditCardField.svelte';
import { render, fireEvent } from '@testing-library/svelte';

/**
 * Prefix of testid attribute for the input element
 */
const inputPrefix: string = "input";
/**
 * Prefix of testid attribute for the error div element
 */
const errorDivPrefix: string = "errorDiv";

/**
 * Properties to render the component as test data set
 */
let options: any = {
  fieldName: "1", filedIconClass: "iconclass", fieldPlaceHolder: "placeHolder", fieldValue: "fieldValue",
  fieldError: "", fieldBlur: undefined
};

/**
 * it expects default values for ValidatorCreditCardEditor attributes
 */
it('check constructor', function (): void {
  const rendered = render(CreditCardField, options);
  expect(rendered).toMatchSnapshot();
});

/**
 * It mocks the handle onblur function to see if it is called
 * It fires user blur on the input
 * It expects the mock function to have been called
 */
it('check onblur', async function (): Promise<void> {
  let handleBlur: jest.Mock<any, any> = jest.fn();
  options.fieldBlur = handleBlur;
  const { getByTestId } = render(CreditCardField, options);
  const input: HTMLElement = getByTestId(inputPrefix + "1");
  await fireEvent.blur(input);
  expect(handleBlur).toHaveBeenCalledTimes(1);
});

/**
 * It sets fieldError property to a not empty string value
 * It expects error div element to be rendered
 */
it('check errorDiv in constructor', function (): void {
  options.fieldError = "it is needed";
  options.fieldBlur = undefined;
  options.fieldName = "2";
  const { getByTestId } = render(CreditCardField, options);
  const errorDiv: HTMLElement = getByTestId(errorDivPrefix + "2");
  expect(errorDiv).toBeInTheDocument();
});
