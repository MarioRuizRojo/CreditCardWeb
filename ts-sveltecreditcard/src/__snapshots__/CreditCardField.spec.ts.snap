// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check constructor 1`] = `
Object {
  "component": CreditCardField {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        "fieldValue",
        "1",
        "iconclass",
        "placeHolder",
        "",
        undefined,
        [Function],
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_disconnect": Array [],
      "on_mount": Array [],
      "props": Object {
        "fieldBlur": 5,
        "fieldError": 4,
        "fieldName": 1,
        "fieldPlaceHolder": 3,
        "fieldValue": 0,
        "filedIconClass": 2,
      },
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="input-group input-group-lg mb-3"
        data-testid="maindiv1"
      >
        <span
          class="input-group-text"
        >
          <i
            class="iconclass"
            data-testid="icon1"
          />
        </span>
         
        <input
          class="form-control"
          data-testid="input1"
          name="1"
          placeholder="placeHolder"
          type="text"
        />
         
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
